### ISSUE TITLE
Required arrays properties are being pruned if empty

### ISSUE THREAD

**Issue by cmcnamara87** [json-schema-faker/json-schema-faker#232]
JSON Schema: http://json-schema-faker.js.org/#gist/522be50fb0c576a8b033728cbf0ba089

Result: 
```
{}
```

Expected Results:
```
{
  "myArray": [
    [
      {
        "id": "84f0e9db-251b-4499-b12c-6a3858375389"
      }
    ]
  ]
}
```

I'd like to be able to say, this array (myArray) property will always exist, it might be empty, it might have some number of values. I have set the array property as "required" which should keep it there, regardless of it being empty or not. 

However, the pruning (I think from #123) is removing empty arrays even if they are set as required.

Any ideas?

**Comment by pateketrueke** [comment]
I think we can enable a setting for keeping empty arrays as is.

Of course we need to explore the implications, what do you think?

**Comment by cmcnamara87** [comment]
A setting would be good, but in addition, I think if a property is set as required it should never be pruned.

**Comment by woodenconsulting** [comment]
+1

**Comment by pateketrueke** [comment]
Fully fixed on `0.5.0-rc5`, thanks for your feedback! :beers:
